cql3/expr/expression.cc

```diff
@@ -793,7 +775,11 @@ static value_set possible_lhs_values(const column_definition* cdef,
                             }
                             return unbounded_value_set;
                         },
-                        [&] (token) -> value_set {
+                        [&] (const function_call& token_fun_call) -> value_set {
+                            if (!is_partition_token_for_schema(token_fun_call, *table_schema_opt)) {
+                                on_internal_error(expr_logger, "possible_lhs_values: function calls are not supported as the LHS of a binary expression");
```

Comment:
Maybe add the words "... except token() ..." to this message, since this sort of function?
I also had another thought - maybe it will be nicer to check three cases:
1. If token(p1,p2,p3), it's supported
2. If token(something,else), print an error that only token of full partition key is allowed in LHS
3. If someotherfunction(), print an error that someotherfunction() is not allowed in LHS, the only function allowed is token().

Reply:
It's an `on_internal_error` message, so it is never really meant to be printed. I don't know if there's any reason to make it pretty - once it triggers someone will go and have to do some debugging anyway, and the message is unique, so it's obvious where the error comes from.